File: src/PascalsTriangleII.java
import java.util.ArrayList;
import java.util.List;

public class PascalsTriangleII {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> result = new ArrayList<>();
        result.add(1);
        for (long index = 1, value = 1; index < rowIndex + 1 ; index++) {
            value = (value * (rowIndex - index + 1)) / (index);
            result.add((int) value);
        }
        return result;
    }
}

File: python/pascals_triangle_ii.py
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        result = [1] * (rowIndex + 1)
        val = 1
        for i in range(1, rowIndex):
            val *= (rowIndex + 1 - i)
            val //= i
            result[i] = val
        return result
